app.py

```diff
@@ -211,6 +211,24 @@ def _launch_app(self, context, file_to_open=None, version=None):
         """
         Launches an app
         """
+        try:
```

Comment:
It might be nice to move this piece of code to a decorator in core. I have an inkling it might be reused elsewhere.